src/index.js

```diff
@@ -670,23 +670,54 @@ class LocalstackPlugin {
       }
```

Comment:
nit: Perhaps we could also introduce a global `resolvedHostname` variable to cache the result of the connection check, like we did here: https://github.com/localstack/aws-cdk-local/blob/main/bin/cdklocal#L27-L33 . This could help as a tiny performance enhancement, not having to run the check multiple times 👍